### PR TITLE
refactor: Rename repo and pointers to sdk-java

### PR DESCRIPTION
**Pull request by fabriziodemaria**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 !.git*
 /target/
 *.iml
+**dependency-reduced-pom.xml
```
Comment (nicklasl) [inline .gitignore:8]: 🙌

File: README.md [hunk 2]
```diff
@@ -38,34 +30,31 @@ To use a snapshot, add the following repository to your `pom.xml`:
...
-import com.spotify.confidence.ConfidenceFeatureProvider;
-import dev.openfeature.sdk.Client;
-import dev.openfeature.sdk.MutableContext;
-import dev.openfeature.sdk.OpenFeatureAPI;
-import dev.openfeature.sdk.Value;
-import java.util.Map;
-
-public final class ResolveFlags {
+final Confidence confidence = Confidence.builder("<CLIENT_TOKEN>").build();
+confidence.setContext(Map.of("country", ConfidenceValue.of("SE")));
```
Comment (nicklasl) [inline README.md:40]: I would actually advocate for people using something like this:
```

final String propertyValue = confidence.withContext(Map.of(
                "user_id", ConfidenceValue.of("<some-user-id>"), 
                "country", ConfidenceValue.of("SE"))
        ).getValue(
    "flag-name.property-name",
    "defaultValue"
);
```

At least we should not use `country` as an example for mutating the root context.

File: pom.xml [hunk 3]
```diff
@@ -14,11 +14,11 @@
   </distributionManagement>
 
   <groupId>com.spotify.confidence</groupId>
-  <artifactId>openfeature-provider</artifactId>
+  <artifactId>sdk-java</artifactId>
   <version>0.0.12-SNAPSHOT</version>
 
-  <url>https://github.com/spotify/confidence-openfeature-provider-java</url>
-  <description>Confidence OpenFeature provider for Java</description>
+  <url>https://github.com/spotify/sdk-java</url>
```
Comment (nicklasl) [inline pom.xml:20]: This is not correct.

File: pom.xml [hunk 4]
```diff
@@ -35,32 +35,32 @@
...
+  <name>Confidence SDK</name>
   <organization>
       <name>com.spotify</name>
       <url>https://github.com/spotify</url>
   </organization>
 
   <issueManagement>
       <system>GitHub</system>
-      <url>https://github.com/spotify/confidence-openfeature-provider-java/issues</url>
+      <url>https://github.com/spotify/sdk-java/issues</url>
```
Comment (nicklasl) [inline pom.xml:46]: Incorrect.

File: pom.xml (hunk 3 above)
Comment (fabriziodemaria) [reply pom.xml:20]: Ops, good catch!

File: pom.xml [hunk 5]
```diff
@@ -35,32 +35,32 @@
...
           <distribution>repo</distribution>
       </license>
   </licenses>
 
   <scm>
-      <url>https://github.com/spotify/confidence-openfeature-provider-java</url>
+      <url>https://github.com/spotify/confidence-sdk-java</url>
       <connection>
-          scm:git:git@github.com:spotify/confidence-openfeature-provider-java.git
+          scm:git:git@github.com:spotify/sdk-java.git
```
Comment (nicklasl) [inline pom.xml:60]: I am not exactly sure what these scm uris are used for but they should probably be set to `confidence-sdk-java`
```suggestion
          scm:git:git@github.com:spotify/confidence-sdk-java.git
```

File: pom.xml [hunk 6]
```diff
@@ -35,32 +35,32 @@
...
       </connection>
       <developerConnection>
-          scm:git:git@github.com:spotify/confidence-openfeature-provider-java.git
+          scm:git:git@github.com:spotify/sdk-java.git
```
Comment (nicklasl) [inline pom.xml:63]:
```suggestion
          scm:git:git@github.com:spotify/confidence-sdk-java.git
```

**Pull request merged**